File: php/wordy/wordy.php
<?php
function calculate(string $q){
    // 1. Patterns
    $num = "(-?\d+)";
    $pats = [
        "/^What is $num/" => "_$1",
        "/plus $num/" => "+_$1",
        "/minus $num/" => "-_$1",
        "/multiplied by $num/" => "*_$1",
        "/divided by $num/" => "/_$1",
        "/raised to the $num\w+ power/" => "**_$1",
        "/\?$/" => "",
    ];

    // 2. Normalize
    $norm = preg_replace(array_keys($pats), array_values($pats), $q);
    $queue = explode(" ",$norm);
    $total = 0;

    // 3. ???
    foreach($queue as $do){
        $els = explode("_", $do);
        if(count($els) <= 1){
            throw new InvalidArgumentException("");
        }
        list($operator,$val) = explode("_", $do);

        switch($operator){
            case "":
            case "+":
                $total += $val;
                break;
            case "-":
                $total -= $val;
                break;
            case "*":
                $total *= $val;
                break;
            case "/":
                $total /= $val;
                break;
            case "**":
                $total **= $val;
                break;
        }
    }

    // 4. PROFIT !
    return $total;
}

File: python/wordy/wordy.py
import re


def calculate(q):
    # 1. patterns
    num = r"(-?\d+)"
    pats = [
        (f"^What is {num}", r"+_\1"),
        (f"plus {num}", r"+_\1"),
        (f"minus {num}", r"-_\1"),
        (f"multiplied by {num}", r"*_\1"),
        (f"divided by {num}", r"/_\1"),
        (f"raised to the {num}\w+ power/", r"**_\1"),
        ("\?$", ""),
    ]

    # 2. normalize
    for pat, repl in pats:
        q = re.sub(pat, repl, q)

    # 3. ???
    total = 0
    for do in q.split(" "):
        operator, val = do.split("_")
        val = int(val)
        total = {
            '+': lambda t: t + val,
            '-': lambda t: t - val,
            '*': lambda t: t * val,
            '/': lambda t: t / val,
            '**': lambda t: t ** val
        }[operator](total)

    # 4. Profit
    return total
